.github/workflows/testgen.yml

```diff
@@ -47,4 +47,6 @@ jobs:
         run: pip install -r requirements.txt
       - name: Generate test files
         working-directory: ./tests
-        run: python3 generate.py
\ No newline at end of file
+        # Run generate with `--force` to ensure freshly generated certs/keys
+        # pass `cargo test`.
+        run: python3 generate.py --force
```

Comment:
If we're going to do this, should we still commit the generated test code? It feels weird to me that we're overwriting the tests here without looking (meaning that the committed code could be out of sync with the generated code).

Honestly I would prefer that we don't regenerate test code in CI and instead make it easy to regenerate only the input certificates when necessary, and maybe make sure that they have the a long lifetime (as long as our library code will allow?).

Reply:
> If we're going to do this, should we still commit the generated test code? It feels weird to me that we're overwriting the tests here without looking (meaning that the committed code could be out of sync with the generated code).

That makes sense. I'd be happy to drop the `--force` argument here. I don't think it adds much.

> Honestly I would prefer that we don't regenerate test code in CI and instead make it easy to regenerate only the input certificates when necessary

I think this will require another round of substantial refactoring to the Python generation code and I'm not super keen to keep investing time in that as opposed to making progress on landing CRL support end to end. My light preference would be to return to that post-CRLs and spend the energy rewriting the whole shebang in Rust. Do you think that's reasonable?

> maybe make sure that they have the a long lifetime (as long as our library code will allow?).

I'm not sure I understand you here. Can you elaborate on what you had in mind?

Reply:
#97 addresses my concerns for now.

> I'm not sure I understand you here. Can you elaborate on what you had in mind?

I was under the impression that we regenerate the DER files in order to update their validity windows. If we can extend the validity windows that alleviate the need to regularly regenerate these files.

Reply:
> https://github.com/rustls/webpki/pull/97 addresses my concerns for now.

:+1: Ok great.

> I was under the impression that we regenerate the DER files in order to update their validity windows. If we can extend the validity windows that alleviate the need to regularly regenerate these files.

From my perspective I thought having the code to generate the certs in-tree was more to make it easier to show that the tests were doing what we expect (e.g. reading the cert generation code vs looking at opaque DER) and to ensure that if the tests need updating we don't have to do it from scratch not knowing how the original DER was made.

The passage of time w.r.t the validity window isn't a problem because the test drivers use a fixed point in time for validation:
https://github.com/rustls/webpki/blob/1d1f05a3c777365b2ebd4b36235cf001ca2e6cca/tests/client_auth.rs#L35

That point in time is kept within the validity range used by the generator:
https://github.com/rustls/webpki/blob/1d1f05a3c777365b2ebd4b36235cf001ca2e6cca/tests/generate.py#L31-L32

Reply:
> Agree with all of that, but those are not reasons to run the cert generation code in CI, right?

Yup! I think we're on the same page here. I agree that generating the certs in CI wasn't the right call.